### PR TITLE
Homebrew-on-Linux.md: update minimal requirements

### PR DESCRIPTION
**Pull request by iMichka**
These are needed to build patchelf 0.10 with debian wheezy.

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: docs/Homebrew-on-Linux.md [hunk 1]
```diff
@@ -47,9 +47,9 @@ If you're using an older distribution of Linux, installing your first package wi
 
 ## Linux/WSL Requirements
 
-+ **GCC** 4.4 or newer
++ **GCC** 4.7.2 or newer
```
Comment (sjackman) [inline docs/Homebrew-on-Linux.md]:
```suggestion
+ **GCC** 4.7.0 or newer
```

Comment (iMichka) [reply docs/Homebrew-on-Linux.md]: I did not test 4.7.0. Not sure which glibc/gcc combination can be found out there. Old distributions have probably updated to the latest minor version of gcc, so I hope most of the people will have 4.7.2. But I may be wrong.

Comment (sjackman) [reply docs/Homebrew-on-Linux.md]: The minimum GCC version required to build glibc 2.23 is 4.7.0. That's the reason for this minimum version.

Comment (sjackman) [reply docs/Homebrew-on-Linux.md]: See the preamble at https://github.com/Linuxbrew/brew/wiki/Bottles

**Pull request merged**